src/main/java/org/zalando/nakadi/controller/EventPublishingController.java

```diff
@@ -114,6 +104,21 @@ private ResponseEntity postEventInternal(final String eventTypeName,
         }
     }
 
+    private void reportMetrics(final EventTypeMetrics eventTypeMetrics, final EventPublishResult result,
+                               final String eventsAsString, final int eventCount) {
+        if (result.getStatus() == EventPublishingStatus.SUBMITTED) {
+            eventTypeMetrics.reportSizing(eventCount, eventsAsString.getBytes(StandardCharsets.UTF_8).length - 2);
+        } else if (result.getStatus() == EventPublishingStatus.FAILED && eventCount != 0) {
+            final int successfulEvents= result.getResponses()
+                    .stream()
+                    .filter(r -> r.getPublishingStatus() == EventPublishingStatus.SUBMITTED)
+                    .collect(Collectors.toList())
+                    .size();
+            final double avgEventSize = eventsAsString.getBytes(StandardCharsets.UTF_8).length / eventCount;
```

Comment:
I think it will not help. It will divide int by int and the result will be int and then it will be converted to double.
So if length is 10 and eventCount is 3, the avgEventSize will be 3.0